src/gangliad/statsd.cpp

```diff
@@ -987,21 +1057,138 @@ StatsD::publishMetrics( int /* timerID */ )
 		}
 	}
 
-	CollectorList* collectors = daemonCore->getCollectorList();
-	ASSERT( collectors );
+	// Now that we have formulated the query, go out and query collector(s)
+	
+	// Handle config knob MONITOR_MULTIPLE_COLLECTORS, which is an optional knob to
+	// tell the gangliad to monitor a list of specified collectors instead of the
+	// default pool collector.  This knob should be a comma seperated list of entries,
+	// where each entry is name-of-collector/address-of-collector, where the address is
+	// same as what you would give the "-pool" parameter of condor_status.  For instance,
+	// MONITOR_MULTIPLE_COLLECTORS = ce1.wisc.edu/ce1.wisc.edu:9619, ce2.wisc.edu/ce2.wisc.edu:9619
+	bool multiple_collectors = false;
+	std::vector<std::string> collector_pool_list;
+	std::vector<std::string> collector_name_list;
+	char *param_monitor_multiple_collectors = NULL;
+	if (m_param_monitor_multiple_collectors.empty()) {
+		param_monitor_multiple_collectors = param("MONITOR_MULTIPLE_COLLECTORS");
+	} else {
+		param_monitor_multiple_collectors = strdup(m_param_monitor_multiple_collectors.c_str());
+	}
+	if (param_monitor_multiple_collectors && param_monitor_multiple_collectors[0]) {
+		StringList collist(param_monitor_multiple_collectors);
+		free(param_monitor_multiple_collectors);
+		param_monitor_multiple_collectors = NULL;
+		collist.rewind();
+		char *col;
+		while( (col=collist.next()) ) {
+			char* pos = strchr(col,'/');
+			if (!pos || !pos[1]) {
+				break;
+			}
+			*pos = '\0';
+			collector_name_list.push_back(col);
+			collector_pool_list.push_back(pos+1);
+		}
+		if (collector_name_list.size() > 0 && 
+			collector_name_list.size() == collector_pool_list.size()) 
+		{
+			multiple_collectors = true;
+		} else {
+			EXCEPT("ERROR: config knob MONITOR_MULTIPLE_COLLECTORS malformed");
+		}
+	}
+
+	int num_collectors = multiple_collectors ? collector_pool_list.size() : 1;
+	int num_ads_from_prev_rounds = 0;
+	CollectorList* collectors = nullptr;
+	for (int i=0; i < num_collectors; i++) {
```

Comment:
All the later `collector_pool_list[i]` could then be replaced with `pool` and likewise for `collector_name_list[i]`

Reply:
Ideally probably should have been a vector of tuples instead of two associative vectors (as coded), or a map (as suggested).  But there is nothing intrinsically wrong with the code as-is, and it has already been tested for weeks in production, so I do not want to change it.